### ISSUE TITLE
sleep command interrupted at start

### ISSUE THREAD

**Issue by byllyfish** [byllyfish/shellous#62]
sleep process was started and then killed, but not by me...

Note that output_bytes and exit_code should not be None. Also, there is a 100msec delay between execute program and process created debug messages.

```
-------------------------------- live log setup --------------------------------
1631668914.399 DEBUG Using selector: KqueueSelector
1631668914.399 DEBUG Using selector: KqueueSelector
1631668914.400 DEBUG Close <_UnixSelectorEventLoop running=False closed=False debug=True>
-------------------------------- live log call ---------------------------------
1631668914.427 INFO Runner.__aenter__ stepin <Runner 'sleep' pid=None> (macOS-10.15.7 CPython 3.10.0rc2 asyncio.unix_events._UnixSelectorEventLoop MainThread ThreadedChildWatcher)
1631668914.431 INFO Runner._start stepin <Runner 'sleep' pid=None>
1631668914.432 DEBUG execute program 'sleep' stdin=<pipe> stdout=<pipe> stderr=<Redirect.DEVNULL: -3>
1631668914.546 DEBUG process 'sleep' created: pid 4340
1631668914.548 WARNING Executing <Task pending name='Task-831' coro=<_wrap_awaitable() running at /Users/runner/hostedtoolcache/Python/3.10.0-rc.2/x64/lib/python3.10/asyncio/tasks.py:648> wait_for=<Future pending cb=[Task.task_wakeup()] created at /Users/runner/hostedtoolcache/Python/3.10.0-rc.2/x64/lib/python3.10/asyncio/base_events.py:424> cb=[_release_waiter(<Future pendi...events.py:424>)() at /Users/runner/hostedtoolcache/Python/3.10.0-rc.2/x64/lib/python3.10/asyncio/tasks.py:387] created at /Users/runner/hostedtoolcache/Python/3.10.0-rc.2/x64/lib/python3.10/asyncio/tasks.py:638> took 0.123 seconds
1631668914.550 DEBUG Write pipe 10 connected: (<_UnixWritePipeTransport fd=10 idle bufsize=0>, <WriteSubprocessPipeProto fd=0 pipe=<_UnixWritePipeTransport fd=10 idle bufsize=0>>)
1631668914.551 WARNING Close running child process: kill <_UnixSubprocessTransport closed pid=4340 running stdin=<_UnixWritePipeTransport closing fd=10 idle bufsize=0>>
1631668914.552 DEBUG process 4340 exited with returncode -9
1631668914.553 DEBUG Read pipe 11 connected: (<_UnixReadPipeTransport fd=11 polling>, <ReadSubprocessPipeProto fd=1 pipe=<_UnixReadPipeTransport fd=11 polling>>)
1631668914.554 INFO <_UnixSubprocessTransport closed pid=4340 running stdin=<_UnixWritePipeTransport closed fd=10 closed> stdout=<_UnixReadPipeTransport fd=11 polling>> exited with return code -9
1631668914.554 INFO <_UnixReadPipeTransport fd=11 polling> was closed by peer
1631668914.555 INFO _RunOptions.exit 'sleep' exc_value=CancelledError()
1631668914.555 INFO Runner.start <Runner 'sleep' pid=None> ex=CancelledError()
1631668914.555 INFO Runner._start stepout <Runner 'sleep' cancelled pid=None> ex=CancelledError()
1631668914.555 INFO Runner.__aenter__ stepout <Runner 'sleep' cancelled pid=None> ex=ResultError(Result(output_bytes=None, exit_code=None, cancelled=True, encoding='utf-8', extra=None))
FAILED
------------------------------ live log teardown -------------------------------
1631668914.706 DEBUG Close <_UnixSelectorEventLoop running=False closed=False debug=True>

```

**Comment by byllyfish** [comment]
This is the code in asyncio.base_events.py which logs the "execute program" message. Note that the second INFO message does NOT appear, which indicates that _make_subprocess_transport was cancelled.

```
        if self._debug:
            # don't log parameters: they may contain sensitive information
            # (password) and may be too long
            debug_log = f'execute program {program!r}'
            self._log_subprocess(debug_log, stdin, stdout, stderr)
        transport = await self._make_subprocess_transport(
            protocol, popen_args, False, stdin, stdout, stderr,
            bufsize, **kwargs)
        if self._debug and debug_log is not None:
            logger.info('%s: %r', debug_log, transport)
```

**Comment by byllyfish** [comment]
Fixed in 0.5.0
